borbskeeper/Features/Profile/ProfileViewController.m

```diff
@@ -0,0 +1,162 @@
+    resizeImageView.image = image;
+    
+    UIGraphicsBeginImageContext(size);
+    [resizeImageView.layer renderInContext:UIGraphicsGetCurrentContext()];
+    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    return newImage;
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+    self.originalImage = info[UIImagePickerControllerOriginalImage];
+    self.editedImage = [self resizeImage:self.originalImage withSize:CGSizeMake(1000, 1000)];
+    [self.profilePicture setImage:self.editedImage];
+    
+    [User currentUser][USER_PROF_PIC_KEY] = [BorbParseManager getPFFileFromImage:self.editedImage];
+    
+    [BorbParseManager saveUser:[User currentUser] withCompletion:nil];
+    
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (IBAction)didTapChangeProfilePicture:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+    } else {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    }
+    [self presentViewController:imagePickerVC animated:YES completion:nil];
+}
+
+- (void)refreshCompleteTaskList{
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(fetchData) forControlEvents:UIControlEventValueChanged];
+    [self.completedTasksListInfiniteScrollView.tableView insertSubview:self.refreshControl atIndex:0];
+    [self.completedTasksListInfiniteScrollView.tableView addSubview:self.refreshControl];
+    [self.activityIndicator startAnimating];
+}
+
+- (void)fetchData {
+    [BorbParseManager fetchCompleteTasksOfUser:User.currentUser.username WithCompletion:^(NSMutableArray *tasks) {
+        self.completeTaskList = tasks;
+        [self.completedTasksListInfiniteScrollView.tableView reloadData];
+        [self.activityIndicator stopAnimating];
+        [self.refreshControl endRefreshing];
+        [self checkDate];
+    }];
+}
+
+- (void)loadMoreData{
+    Task *latestTask = [self.completeTaskList lastObject];
+    self.latestDate = latestTask.createdAt;
+    
+    [BorbParseManager loadMoreCompleteTasksOfUser:User.currentUser.username withLaterDate:self.latestDate WithCompletion:^(NSMutableArray *posts) {
+        if ([posts count] > 0){
+            [self.completeTaskList addObjectsFromArray:posts];
+            [self.completedTasksListInfiniteScrollView.tableView reloadData];
+            self.completedTasksListInfiniteScrollView.isMoreDataLoading = false;
+        } else {
+            self.completedTasksListInfiniteScrollView.isMoreDataLoading = true;
```

Comment:
why do we set this to true?

Reply:
So that you can't pull down to load more data when there is none, instead giving the user a smoother bounce effect at bottom.